docs/framework/30scoring/implementation.md

```diff
@@ -1,51 +1,186 @@
+[Mixing of Control and Data Plane](../20categories/30implementation/mixing-control-data-flow.md) section. If the Control and Data Plane are mixed, then the Mixing of Control and Data Plane Risk Score is 20, or 0 otherwise.
+
+The Control and Data Plane are considered mixed if the answer to any of the following questions is yes:
+
+* Can Control Plane functions be called from Data Plane functions without some additional authentication checks?
+* If Control Plane messages are sent across the crosschain communications channel, are Control Plane messages signed by the same keys as Data Plane messages?
+
+
+#### Role Based Access Control Risk Score
+Role Based Access Control is described in the [Role Based Access Control](../20categories/30implementation/access-control.md) section.
+
+The Role Based Access Control Risk Score depends on the permissioned control / ownership of the bridge contracts. If there are multiple contracts, and they  are configured differently, then score the highest score.
+
+* There is no permissioned ownership feature: Score 0.
+* There a single EOA as the owner: Score 10.
+* Role Based Access Control has been implemented: Score 0.
+* There is no Role Based Access Control, but a threshold number of signers need to approve administrative actions: Score 5.
+
+
+
+#### Upgradable Risk Score
+Upgradable Risk is decribed in the [Upgradability](../20categories/30implementation/upgrade.md) section.
+
+This is currently not accounted for and should be scored as 0.
+
+#### Secret Storage Risk Score
+Secret Storage is described in the [Secret Storage](../20categories/30implementation/secret-storage.md) section.
+
+This is currently not accounted for and should be scored as 0.
+
+#### Product Development Maturity Risk Score
+Product Development Maturity is described in the [Product Development Maturity](../20categories/30implementation/maturity.md) section.
+
+
+This is currently not accounted for and should be scored as 0.
+
+#### Well Known Platform Risk Score
+The risks associated with not using Well Known Platforms is contained in the [Well Known Platform](../20categories/30implementation/known-platform.md) section.
+
+This is currently not accounted for and should be scored as 0.
+
+
+#### Well Known Smart Contract Programming Language Risk Score
+The risks associated with not using a Well Known Smart Contract Programming Language is contained in the [Well Known Smart Contract Programming Language](../20categories/30implementation/known-language.md) section. The Well Known Smart Contract Programming Language Risk Score is specific to a bridge between two chains. That is, a bridge system could be between a dozen chains. The score could be different for each source chain - target chain combination. The Well Known Smart Contract Programming Language Risk Score is calculated as:
 
 ```
-Implementation Risk Score = Code Auditing Score * 100 / 20
+Well Known Smart Contract Programming Language Risk Score = I700 + I701
```

Comment:
I can't seem to find an `I700` should the numbering in the table below be updated or the operand in this sum?

Reply:
Yep. The numbers in the table were wrong. Fixed.